src/net.cpp

```diff
@@ -718,10 +718,11 @@ CNetMessage V1TransportDeserializer::GetMessage(const CMessageHeader::MessageSta
 
     msg.m_valid_checksum = (memcmp(hash.begin(), hdr.pchChecksum, CMessageHeader::CHECKSUM_SIZE) == 0);
     if (!msg.m_valid_checksum) {
-        LogPrint(BCLog::NET, "CHECKSUM ERROR (%s, %u bytes), expected %s was %s\n",
+        LogPrint(BCLog::NET, "CHECKSUM ERROR (%s, %u bytes), expected %s was %s, peer=%d\n",
```

Comment:
test-nit in commit 2616399cca2e9c1a4c03605241c7964d53b360bf

Could extend the test?

```diff
diff --git a/test/functional/p2p_invalid_messages.py b/test/functional/p2p_invalid_messages.py
index d9a9ae5188..04d86d3bc5 100755
--- a/test/functional/p2p_invalid_messages.py
+++ b/test/functional/p2p_invalid_messages.py
@@ -93,7 +93,9 @@ class InvalidMessagesTest(BitcoinTestFramework):
     def test_checksum(self):
         self.log.info("Test message with invalid checksum logs an error")
         conn = self.nodes[0].add_p2p_connection(P2PDataStore())
-        with self.nodes[0].assert_debug_log(['CHECKSUM ERROR (badmsg, 2 bytes), expected 78df0a04 was ffffffff']):
+        log_msg = 'CHECKSUM ERROR (badmsg, 2 bytes), expected 78df0a04 was ffffffff, peer={}'.format(
+            self.nodes[0].getpeerinfo()[0]['id'])
+        with self.nodes[0].assert_debug_log([log_msg]):
             msg = conn.build_message(msg_unrecognized(str_data="d"))
             # Checksum is after start bytes (4B), message type (12B), len (4B)
             cut_len = 4 + 12 + 4

Reply:
This could be done for all of the tests, and can mostly be done separate from the changes in this PR.  I'll do this in another PR.

Reply:
Additionally I'm not sure that we have consensus that `assert_debug_log` is really a good thing.